src/plugins/status_page/public/status_page.js

```diff
@@ -36,6 +36,14 @@ const chrome = require('ui/chrome')
         ui.serverState = overall.state;
         ui.serverStateMessage = overall.title;
       }
+
+      // Unless the status page was explicitly requested, there is no
+      // reason to stay on the status page if the status is green; reload
+      // the window so the user is shown the page they were requesting.
+      const statusPageUrl = chrome.addBasePath('/status');
+      if ((overall.state === 'green') && ($window.location.pathname !== statusPageUrl)) {
+        return $window.location.reload();
+      }
```

Comment:
This bit of code is necessary for the following scenario:

1. There is no Kibana index.
2. Kibana server is started. Elasticsearch plugin is yellow.
3. User tries to open Kibana in the browser before the Kibana index has been created.
4. They are shown the status page (because not all plugins are green yet). However, optimization hasn't completed so they are still shown the loading graphic.
5. Before optimization completes, the Kibana index is created and the Elasticsearch plugin turns green.
6. Optimization completes, and the user is shown the status page, where everything shows green. Unless the user explicitly requested the status page, this is an odd experience IMO. So we reload the page, causing the user to be shown the page they actually requested.